tf03_driver/include/TF03.h

```diff
@@ -0,0 +1,136 @@
+#include <math.h>
+#include <iostream>
+#include <string>
+#include <vector>
+#include <sys/socket.h>
+#include <net/if.h>
+#include <sys/ioctl.h>
+#include <linux/can.h>
+#include <linux/can/raw.h>
+
+
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/msg/range.hpp"
+
+#ifdef LIBSERIAL_0_X
+#include <SerialPort.h>
+#include <SerialStream.h>
+#define SerialPortNotOpen SerialPort::NotOpen
+#define SerialPortReadTimeout SerialPort::ReadTimeout
+#define ReadByteFromSerial(port, buffer, timeout) (buffer = port->ReadByte(timeout))
+#else
+#include <libserial/SerialPort.h>
+#include <libserial/SerialStream.h>
+#define SerialPortNotOpen LibSerial::NotOpen
+#define SerialPortReadTimeout LibSerial::ReadTimeout
+#define ReadByteFromSerial(port, buffer, timeout) (port->ReadByte(buffer, timeout))
+#endif
+
+#ifndef TF03_H
+#define TF03_H
+
+#define SET_OUTPUT_FORMAT_SERIAL 0x01
+#define SET_OUTPUT_FORMAT_CAN 0x02
+
+enum tf_03_interface
+{
+    serial = 0,
+    can = 1
+};
+
+enum tf_03_command_id
+{
+    version = 0x01,
+    system_reset = 0x02,
+    framerate = 0x03,
+    factory_settings = 0x10,
+    save_settings = 0x11,
+    output_format = 0x45,
+    transmit_can_id = 0x50,
+    receive_can_id = 0x51,
+};
+
+struct parameter_config
+{
+    bool frame_sent;
+    bool command_success;
+    tf_03_command_id command;
+    int64_t argument;
+};
+
+class TF03 : public rclcpp::Node
+{
+
+//     SubscriberPublisher()
+//   : Node("minimal_subscriber")
+
+
+public:
+    // TF03();
+    // ~TF03();
+   TF03(); //: Node("tf03driver");
```

Comment:
Default empty constructor that is not used should be either refactored or entirely removed. It is the best to fill it with necessery content in implementation later on so let it be but it is a general advice